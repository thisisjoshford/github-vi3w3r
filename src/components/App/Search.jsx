  
import { React } from 'react';
import PropTypes from 'prop-types';

const Search = ({ userName, onUserChange, onUserSubmit }) => {
  <form onSubmit={onUserSubmit}>
    <input 
      type="text" 
      name="userName" 
      value={userName} 
      onChange={onUserChange} />
    <button 
      name="search"
      value="search"
      onClick={onUserSubmit}
    >Search
    </button>
  </form>;
 
};

Search.propTypes = {
  userName: PropTypes.string.isRequired,
  onUserChange: PropTypes.func.isRequired,
  onUserSubmit: PropTypes.func.isRequired
};

export default Search;
