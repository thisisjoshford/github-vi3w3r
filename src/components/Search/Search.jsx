import React from 'react';
import PropTypes from 'prop-types';

const Search = ({ userName, onUserChange, onUserSubmit }) => (
  <>
    <h1>Github Viewer</h1>
    <input 
      type="text" 
      name="userName" 
      value={userName} 
      onChange={onUserChange} 
    />
    <input 
      type="submit"
      name="search"
      value="Search"
      onClick={onUserSubmit}
    />
  </>
 
);

Search.propTypes = {
  userName: PropTypes.string.isRequired,
  onUserChange: PropTypes.func.isRequired,
  onUserSubmit: PropTypes.func.isRequired
};

export default Search;
