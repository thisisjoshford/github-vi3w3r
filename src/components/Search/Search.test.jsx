import React from 'react';
import { shallow } from 'enzyme';
import Search from './Search';

describe('Search component', () => {
  it('renders Search', () => {
    const wrapper = shallow(<Search
      userName="thisisjoshford"
      onUserChange={() => {}}
      onUserSubmit={() => {}} 
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
