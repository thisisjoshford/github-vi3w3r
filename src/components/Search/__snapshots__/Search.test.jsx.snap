// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders Search 1`] = `
<Fragment>
  <input
    name="userName"
    onChange={[Function]}
    type="text"
    value="thisisjoshford"
  />
  <input
    name="search"
    onClick={[Function]}
    type="submit"
    value="Search"
  />
</Fragment>
`;
